src/leafdoc.js

```diff
@@ -395,17 +402,18 @@ Leafdoc.prototype.addStr = function(str, isSource) {
 
 // 						console.log(content, ', ', alt);
 
-					var name, params = {}, type = null, defaultValue = null;
+					var name, params = {}, type = null, defaultValue = null, optional = false;
 
 					if (content) {
 						var split = regexps.functionDefinition.exec(content);
 						if (!split) {
 							console.error('Invalid ' + directive + ' definition: ', content);
 						} else {
 							name = split[1];
-							paramString = split[2];
-							type = split[3];
-							defaultValue = split[4];
+							optional = split[2] == '?';
```

Comment:
Sure. Suppose we have some data structures for our api. And I want provide default value and optionality:

```html
<table><thead>
	<tr>
		<th>Field</th>
		<th>Type</th>
		<th>Required</th>
		<th>Example</th>
		<th>Description</th>
	</tr>
	</thead><tbody>
	{{#each documentables}}
	<tr id='{{id}}'>
		<td><code><b>{{name}}</b></code></td>
		<td><code>{{{type type}}}</code>
		<td>{{#unless optional}}Yes{{/unless}}</td>
		<td><code>{{defaultValue}}</code></td>
		<td>{{{markdown comments}}}</td>
	</tr>
	{{/each}}
</tbody></table>
```